File: 0801_0900/0877/0877.py
with open("input.txt", "r") as f, open("output.txt", "w") as q:
    name = f.readline().split()[1]
    s = f.readline().strip()[10:]
    d = False
    while s != name + " signed off":
        if s[-1] == '.':
            s = s[:-1] + ','
        elif s[-1] not in '!?':
            s += ','
        q.write(f'"{s}" --- skazal {name if d else "Fedya"}.\n')
        d = not d
        s = f.readline().strip()[10:]

File: 0801_0900/0877/0877.c
#include <stdio.h>
#include <string.h>

int main()
{
	FILE *f = fopen("input.txt", "r"), *q = fopen("output.txt", "w");
	int n;
	char s[251], name[230];

	fscanf(f, "%s%s%s%s", s, name, s, s);
	n = strlen(name);
	for (int i = 0, k; fscanf(f, "%s %[^\n]\n", s, s) && (strncmp(s, name, n) || strcmp(s + n, " signed off")); ++i)
	{
		if (s[(k = (strlen(s) - 1))] == '.')
			s[k] = ',';
		else if (s[k] != '!' && s[k] != '?')
			s[++k] = ',', s[++k] = '\0';
		fprintf(q, "\"%s\" --- skazal %s.\n", s, i % 2 ? name : "Fedya");
	}
	return 0;
}
